### PR TITLE
Avoid dependency injection for template host logging

### PR DESCRIPTION
**Pull request by JeremyKuhne**
## Summary

`CliTemplateEngineHost` currently builds a dependency injection container each time it creates its logger factory. Construct the existing console provider and formatter directly instead.

- Preserve the requested minimum log level, scopes, timestamp format, exception formatting, and console-provider defaults.
- Register the provider with `LoggerFactory.AddProvider` so disposing the host continues to own and drain the asynchronous console provider.
- Use fixed options monitors because this host has no configuration source capable of changing these options after construction.
- Add focused tests for every minimum log level, formatter behavior, scopes, exception output, repeated host lifetimes, and queued-output drain on disposal.

## Performance

The host constructor's sampled CPU cost fell from 15.15 ms to 4.44 ms per invocation, a reduction of 10.71 ms. A separate host construction/disposal allocation measurement fell from 93,191 to 9,801 bytes per host (89.5%).

Windows x64 Release ReadyToRun measurements used 15 alternating before/after pairs per command:

| Command | Before median | After median | Median paired improvement |
| --- | ---: | ---: | ---: |
| `dotnet new --help` | 151.21 ms | 137.23 ms | 9.23% |
| `dotnet new list` | 196.71 ms | 178.56 ms | 8.56% |
| `dotnet new console --help` | 188.56 ms | 176.99 ms | 5.47% |
| `dotnet new console --no-restore` | 248.86 ms | 238.03 ms | 5.32% |

Percentages are medians of paired changes rather than ratios of the separately reported arm medians.

## Testing

- `CliTemplateEngineHostTests`: 8 passed in Debug
- `CliTemplateEngineHostTests`: 8 passed in Release
- Focused `dotnet-new.IntegrationTests` list/help/create slice: 6 passed in Release
- `build.cmd -c Release`

### CODE REVIEW

File: src/Cli/Microsoft.TemplateEngine.Cli/CliTemplateEngineHost.cs [hunk 1]
```diff
@@ -48,6 +36,37 @@ public CliTemplateEngineHost(
...
+
+        private sealed class StaticOptionsMonitor<TOptions>(TOptions options)
+            : IOptionsMonitor<TOptions> where TOptions : new()
+        {
+            public TOptions CurrentValue => options;
+
+            public TOptions Get(string? name) => options;
+
+            public IDisposable? OnChange(Action<TOptions, string?> listener) => null;
+        }
```
[the remark this answers is not in the document]
Comment (JeremyKuhne) [reply src/Cli/Microsoft.TemplateEngine.Cli/CliTemplateEngineHost.cs:68]: `IOptionsMonitor<T>.OnChange` returns `IDisposable?` in the current contract. Returning `null` is supported for a monitor that never changes; the console logger's internal `FormatterOptionsMonitor<T>` follows the same pattern. Both consumers also use null-conditional disposal, so this cannot cause the stated `NullReferenceException`.

**Review by joeloff — APPROVED** [review]
